### ISSUE TITLE
Script requires charset to be set to utf-8

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor.js#314]
Since the addition of the extended regular expression character classes, the Asciidoctor.js script now requires that the charset to be set to utf-8. When loading the script in an HTML page, either the charset must be declared in the meta tag:

```html
<meta charset="utf-8">
```

or it must be declared in the script tag:

```html
<script src="asciidoctor.js" charset="utf-8"></script>
```

I'm not sure if this is a good thing or not. The reason we define the character classes using Unicode characters is to save bytes. However, it might be safer if we keep the script charset independent and use escape codes instead. We should decide before releasing 1.5.6.

**Comment by mojavelinux** [comment]
Here's the error if the charset is not properly set:

```
Uncaught SyntaxError: Invalid regular expression: /^([A-Za-z...
```

**Comment by Mogztter** [comment]
> I'm not sure if this is a good thing or not.

Using utf-8 charset is a good thing I guess.

> However, it might be safer if we keep the script charset independent and use escape codes instead.

True... 

> The reason we define the character classes using Unicode characters is to save bytes

In this case, I think we can avoid using Unicode characters because a few more bytes is not a big deal (at least in Asciidoctor.js) compared to a runtime error if this new requirement is not met.

**Comment by mojavelinux** [comment]
I'd definitely like to get this corrected before 1.5.6. Now that we proved it works, I think we should add a utility task to the build that generates the ranges used in the rx.rb file. That way, we can updated it when needed.

**Comment by Mogztter** [comment]
It's already fixed by https://github.com/asciidoctor/asciidoctor.js/issues/43 right ?

As far as I understand, babel-plugin-utf-8-regex is nothing more than a mapping table between UTF-8 RegExp and Unicode ?

> Now that we proved it works, I think we should add a utility task to the build that generates the ranges used in the rx.rb file.

Could you please explain how we can do it ?

**Comment by mojavelinux** [comment]
Even though this can be fixed by setting the charset in the page, it's going to be perceived as an error in Asciidoctor.js. Therefore, I still think it's worth fixing.

I tried to limit the size of the rx.rb by using the non-latin unicode glyphs directly. The downside is that this requires the file be read as UTF-8. What we need to do instead is use the escape sequences (like XRegExp does). The file ends up being larger, but at least we avoid parsing errors.

For example, instead of:

```
À-Ö
```

we use:

```
\u00c0-\u00d6
```

Of course, the easiest way to get these ranges is to go back to the source of how I made them in the first place. After getting the ranges, I had pasted them into the JavaScript console, copied the resolved glyphs, then paste them into the Ruby file. We just need to skip that resolution step.
